src/server/project.js

```diff
@@ -0,0 +1,346 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+
+var fs = require('fs'),
+    path = require('path'),
+    cordovaServe = require('cordova-serve'),
+    Q = require('q'),
+    dirs = require('./dirs'),
+    pluginUtil = require('./utils/plugins'),
+    prepareUtil = require('./utils/prepare'),
+    utils = require('./utils/jsUtils');
+
+/**
+ * @constructor
+ */
+function Project(simulator, platform) {
+    this._simulator = simulator;
```

Comment:
I'm thinking 'simulatorProxy` is the wrong terminology here. That would imply this is some sort of intermediary between the simulator and other things. But really it is more of a "context" type object, right? How about `simulatorContext` or `simulationContext`?